I need to enable `$(AndroidGenerateResourceDesigner)` in various test projects:
```
/Users/builder/azdo/_work/1/s/src/Compatibility/ControlGallery/src/Android/CustomRenderers.cs(429,46): error CS0117: 'Resource.Id' does not contain a definition for 'Image'
```

--- a/DEVELOPMENT.md
+++ b/DEVELOPMENT.md
@@ -71,3 +71,32 @@ To build and run WinUI 3 support, please install the additional components menti
 dotnet tool restore
 dotnet cake --target=VS-WINUI
 ```
+
+### Android
+
+To workaround a performance issue, all `Resource.designer.cs`
+generation is disabled for class libraries in this repo.
+
+If you need to add a new `@(AndroidResource)` value to be used from C#
+code in .NET MAUI:
+
+1. Comment out the `<PropertyGroup>` in `Directory.Build.targets` that
+   sets `$(AndroidGenerateResourceDesigner)` and
+   `$(AndroidUseIntermediateDesignerFile)` to `false`.
+
+2. Build .NET MAUI as you normally would. You will get compiler errors
+   about duplicate fields, but `obj\Debug\net6.0-android\Resource.designer.cs`
+   should now be generated.
+
+3. Open `obj\Debug\net6.0-android\Resource.designer.cs`, and find the
+   field you need such as:
+
+```csharp
+// aapt resource value: 0x7F010000
+public static int foo = 2130771968;
+```
+
+4. Copy this field to the `Resource.designer.cs` checked into source
+   control, such as: `src\Controls\src\Core\Platform\Android\Resource.designer.cs`
+
+5. Restore the commented code in `Directory.Build.targets`.

--- a/Directory.Build.targets
+++ b/Directory.Build.targets
@@ -6,6 +6,15 @@
   <Import Project="eng\AndroidX.targets" />
   <Import Project="eng\Microsoft.Extensions.targets" />
 
+  <!--
+    Comment out this section if you need to update Resource.designer.cs files.
+    See DEVELOPMENT.md#Android for details.
+  -->
+  <PropertyGroup Condition="'$(TargetPlatformIdentifier)' == 'Android' and '$(AndroidApplication)' != 'true'">
+    <AndroidGenerateResourceDesigner>false</AndroidGenerateResourceDesigner>
+    <AndroidUseIntermediateDesignerFile>false</AndroidUseIntermediateDesignerFile>
+  </PropertyGroup>
+
   <!-- HACK: Prevent the Platform checks -->
   <Target Name="BinPlaceBootstrapDll" />
 

--- a/src/Compatibility/Core/src/Android/Resource.designer.cs
+++ b/src/Compatibility/Core/src/Android/Resource.designer.cs
@@ -1,0 +1,143 @@
+#pragma warning disable 1591
+//------------------------------------------------------------------------------
+// This file was manually curated so we could avoid 5,172 fields.
+// See DEVELOPMENT.md#Android for details on updating this file.
+//------------------------------------------------------------------------------
+
+[assembly: global::Android.Runtime.ResourceDesignerAttribute("Microsoft.Maui.Controls.Compatibility.Resource", IsApplication=false)]
+
+namespace Microsoft.Maui.Controls.Compatibility
+{
+	[global::System.CodeDom.Compiler.GeneratedCodeAttribute("Xamarin.Android.Build.Tasks", "12.0.99.31")]
+	public partial class Resource
+	{
+		static Resource()
+		{
+			global::Android.Runtime.ResourceIdManager.UpdateIdValues();
+		}
+
+		public partial class Animation
+		{
+			// aapt resource value: 0x7F01001C
+			public static int enterfromleft = 2130771996;
+
+			// aapt resource value: 0x7F01001D
+			public static int enterfromright = 2130771997;
+
+			// aapt resource value: 0x7F01001E
+			public static int exittoleft = 2130771998;
+
+			// aapt resource value: 0x7F01001F
+			public static int exittoright = 2130771999;
+
+			static Animation()
+			{
+				global::Android.Runtime.ResourceIdManager.UpdateIdValues();
+			}
+		}
+
+		public partial class Attribute
+		{
+			// aapt resource value: 0x7F030007
+			public static int actionBarSize = 2130903047;
+
+			// aapt resource value: 0x7F0300CE
+			public static int collectionViewStyle = 2130903246;
+
+			// aapt resource value: 0x7F030301
+			public static int scrollViewStyle = 2130903809;
+
+			static Attribute()
+			{
+				global::Android.Runtime.ResourceIdManager.UpdateIdValues();
+			}
+		}
+
+		public partial class Drawable
+		{
+			// aapt resource value: 0x7F07001D
+			public static int abc_ic_clear_material = 2131165213;
+
+			// aapt resource value: 0x7F070026
+			public static int abc_ic_search_api_material = 2131165222;
+
+			static Drawable()
+			{
+				global::Android.Runtime.ResourceIdManager.UpdateIdValues();
+			}
+		}
+
+		public partial class Id
+		{
+			// aapt resource value: 0x7F080061
+			public static int bottomtab_navarea = 2131230817;
+
+			// aapt resource value: 0x7F080062
+			public static int bottomtab_tabbar = 2131230818;
+
+			// aapt resource value: 0x7F0800C7
+			public static int flyoutcontent_appbar = 2131230919;
+
+			// aapt resource value: 0x7F0800F4
+			public static int main_tablayout = 2131230964;
+
+			// aapt resource value: 0x7F0800F5
+			public static int main_viewpager = 2131230965;
+
+			// aapt resource value: 0x7F08010C
+			public static int maui_toolbar = 2131230988;
+
+			// aapt resource value: 0x7F08017D
+			public static int shellcontent_toolbar = 2131231101;
+
+			static Id()
+			{
+				global::Android.Runtime.ResourceIdManager.UpdateIdValues();
+			}
+		}
+
+		public partial class Layout
+		{
+			// aapt resource value: 0x7F0B001C
+			public static int bottomtablayout = 2131427356;
+
+			// aapt resource value: 0x7F0B002F
+			public static int flyoutcontent = 2131427375;
+
+			// aapt resource value: 0x7F0B006B
+			public static int rootlayout = 2131427435;
+
+			// aapt resource value: 0x7F0B006F
+			public static int shellcontent = 2131427439;
+			
+			// aapt resource value: 0x7F0B0070
+			public static int shellrootlayout = 2131427440;
+
+			// aapt resource value: 0x7F0B0072
+			public static int tabbar = 2131427442;
+
+			// aapt resource value: 0x7F0B0082
+			public static int toolbar = 2131427458;
+			
+			static Layout()
+			{
+				global::Android.Runtime.ResourceIdManager.UpdateIdValues();
+			}
+		}
+
+		public partial class Style
+		{
+			// aapt resource value: 0x7F0E02DF
+			public static int collectionViewTheme = 2131624671;
+
+			// aapt resource value: 0x7F0E02E1
+			public static int scrollViewTheme = 2131624673;
+
+			static Style()
+			{
+				global::Android.Runtime.ResourceIdManager.UpdateIdValues();
+			}
+		}
+	}
+}
+#pragma warning restore 1591

--- a/src/Controls/src/Core/Platform/Android/Resource.designer.cs
+++ b/src/Controls/src/Core/Platform/Android/Resource.designer.cs
@@ -1,0 +1,140 @@
+#pragma warning disable 1591
+//------------------------------------------------------------------------------
+// This file was manually curated so we could avoid 5,167 fields.
+// See DEVELOPMENT.md#Android for details on updating this file.
+//-----------------------------------------------------------------------------
+
+[assembly: global::Android.Runtime.ResourceDesignerAttribute("Microsoft.Maui.Controls.Resource", IsApplication=false)]
+
+namespace Microsoft.Maui.Controls
+{
+	[global::System.CodeDom.Compiler.GeneratedCodeAttribute("Xamarin.Android.Build.Tasks", "12.0.99.31")]
+	public partial class Resource
+	{
+		static Resource()
+		{
+			global::Android.Runtime.ResourceIdManager.UpdateIdValues();
+		}
+
+		public partial class Animation
+		{
+			// aapt resource value: 0x7F01001C
+			public static int enterfromleft = 2130771996;
+
+			// aapt resource value: 0x7F01001D
+			public static int enterfromright = 2130771997;
+
+			// aapt resource value: 0x7F01001E
+			public static int exittoleft = 2130771998;
+
+			// aapt resource value: 0x7F01001F
+			public static int exittoright = 2130771999;
+
+			static Animation()
+			{
+				global::Android.Runtime.ResourceIdManager.UpdateIdValues();
+			}
+		}
+
+		public partial class Attribute
+		{
+			// aapt resource value: 0x7F030007
+			public static int actionBarSize = 2130903047;
+
+			static Attribute()
+			{
+				global::Android.Runtime.ResourceIdManager.UpdateIdValues();
+			}
+		}
+
+		public partial class Drawable
+		{
+			// aapt resource value: 0x7F07001D
+			public static int abc_ic_clear_material = 2131165213;
+
+			// aapt resource value: 0x7F070022
+			public static int abc_ic_menu_overflow_material = 2131165218;
+
+			// aapt resource value: 0x7F070026
+			public static int abc_ic_search_api_material = 2131165222;
+
+			static Drawable()
+			{
+				global::Android.Runtime.ResourceIdManager.UpdateIdValues();
+			}
+		}
+
+		public partial class Id
+		{
+			// aapt resource value: 0x7F080051
+			public static int appbar = 2131230801;
+
+			// aapt resource value: 0x7F080061
+			public static int bottomtab_navarea = 2131230817;
+
+			// aapt resource value: 0x7F080062
+			public static int bottomtab_tabbar = 2131230818;
+
+			// aapt resource value: 0x7F0800C7
+			public static int flyoutcontent_appbar = 2131230919;
+
+			// aapt resource value: 0x7F0800F4
+			public static int main_tablayout = 2131230964;
+
+			// aapt resource value: 0x7F0800F5
+			public static int main_viewpager = 2131230965;
+
+			// aapt resource value: 0x7F08010C
+			public static int maui_toolbar = 2131230988;
+
+			// aapt resource value: 0x7F08017C
+			public static int shellcontent_appbar = 2131231100;
+
+			// aapt resource value: 0x7F08017D
+			public static int shellcontent_toolbar = 2131231101;
+
+			static Id()
+			{
+				global::Android.Runtime.ResourceIdManager.UpdateIdValues();
+			}
+		}
+
+		public partial class Layout
+		{
+			// aapt resource value: 0x7F0B001C
+			public static int bottomtablayout = 2131427356;
+
+			// aapt resource value: 0x7F0B002F
+			public static int flyoutcontent = 2131427375;
+
+			// aapt resource value: 0x7F0B006E
+			public static int shellcontent = 2131427438;
+
+			// aapt resource value: 0x7F0B006F
+			public static int shellrootlayout = 2131427439;
+
+			static Layout()
+			{
+				global::Android.Runtime.ResourceIdManager.UpdateIdValues();
+			}
+		}
+
+		public partial class String
+		{
+			// aapt resource value: 0x7F0D0065
+			public static int nav_app_bar_navigate_up_description = 2131558501;
+
+			// aapt resource value: 0x7F0D0066
+			public static int nav_app_bar_open_drawer_description = 2131558502;
+
+			// aapt resource value: 0x7F0D0067
+			public static int overflow_tab_title = 2131558503;
+
+			static String()
+			{
+				global::Android.Runtime.ResourceIdManager.UpdateIdValues();
+			}
+		}
+	}
+}
+#pragma warning restore 1591

--- a/src/Core/src/Platform/Android/Resource.designer.cs
+++ b/src/Core/src/Platform/Android/Resource.designer.cs
@@ -1,0 +1,116 @@
+#pragma warning disable 1591
+//------------------------------------------------------------------------------
+// This file was manually curated so we could avoid 5,310 fields.
+// See DEVELOPMENT.md#Android for details on updating this file.
+//-----------------------------------------------------------------------------
+
+[assembly: global::Android.Runtime.ResourceDesignerAttribute("Microsoft.Maui.Resource", IsApplication=false)]
+
+namespace Microsoft.Maui
+{
+	[global::System.CodeDom.Compiler.GeneratedCodeAttribute("Xamarin.Android.Build.Tasks", "12.0.99.31")]
+	public partial class Resource
+	{
+		static Resource()
+		{
+			global::Android.Runtime.ResourceIdManager.UpdateIdValues();
+		}
+
+		public partial class Attribute
+		{
+			// aapt resource value: 0x7F0300E4
+			public static int colorSwitchThumbNormal = 2130903268;
+
+			// aapt resource value: 0x7F030269
+			public static int maui_splash = 2130903657;
+
+			// aapt resource value: 0x7F030301
+			public static int scrollViewStyle = 2130903809;
+
+			static Attribute()
+			{
+				global::Android.Runtime.ResourceIdManager.UpdateIdValues();
+			}
+		}
+
+		public partial class Animation
+		{
+			// aapt resource value: 0x7F010020
+			public static int nav_default_enter_anim = 2130772000;
+			
+			// aapt resource value: 0x7F010021
+			public static int nav_default_exit_anim = 2130772001;
+			
+			// aapt resource value: 0x7F010022
+			public static int nav_default_pop_enter_anim = 2130772002;
+			
+			// aapt resource value: 0x7F010023
+			public static int nav_default_pop_exit_anim = 2130772003;
+
+			static Animation()
+			{
+				global::Android.Runtime.ResourceIdManager.UpdateIdValues();
+			}
+		}
+
+		public partial class Drawable
+		{
+			// aapt resource value: 0x7F07001D
+			public static int abc_ic_clear_material = 2131165213;
+
+			static Drawable()
+			{
+				global::Android.Runtime.ResourceIdManager.UpdateIdValues();
+			}
+		}
+
+		public partial class Id
+		{
+			// aapt resource value: 0x7F080059
+			public static int automation_tag_id = 2131230809;
+
+			// aapt resource value: 0x7F080107
+			public static int maui_toolbar = 2131230983;
+
+			// aapt resource value: 0x7F08012D
+			public static int nav_host = 2131231021;
+
+			// aapt resource value: 0x7F08016A
+			public static int search_button = 2131231082;
+
+			// aapt resource value: 0x7F08016B
+			public static int search_close_btn = 2131231083;
+
+			static Id()
+			{
+				global::Android.Runtime.ResourceIdManager.UpdateIdValues();
+			}
+		}
+
+		public partial class Layout
+		{
+			// aapt resource value: 0x7F0B0059
+			public static int navigationlayout = 2131427417;
+
+			static Layout()
+			{
+				global::Android.Runtime.ResourceIdManager.UpdateIdValues();
+			}
+		}
+
+		public partial class Style
+		{
+			// aapt resource value: 0x7F0E00F8
+			public static int Maui_MainTheme_NoActionBar = 2131624184;
+
+			// aapt resource value: 0x7F0E02E1
+			public static int scrollViewTheme = 2131624673;
+
+			static Style()
+			{
+				global::Android.Runtime.ResourceIdManager.UpdateIdValues();
+			}
+		}
+	}
+}
+#pragma warning restore 1591

